### PR TITLE
Mq messages on error

### LINKED ISSUE #130: Send messages on error
Upon failures that aren't likely local/temporary, we need to make sure we send a message to the correct queue for errors queue for communication to the user as well as nacking the message so we do not get our queues filled up with messages that only lead to failures.

### PR DESCRIPTION
**Pull request by jbygdell**
* Moves JSON validation to a separate function
* Added test for the above function
* Send messges on JSON error
* Send messages on file related errors for ingest and verify

fixes #130

### CODE REVIEW

File: cmd/finalize/finalize.go [hunk 1]
```diff
@@ -79,61 +77,124 @@ func main() {
...
+				josnError := broker.JSONError{
+					Error:          "Handling json message failed",
+					Reason:         err.Error(),
+					OrginalMessage: delivered.Body,
+				}
+				body, _ := json.Marshal(josnError)
+
+				if e := broker.SendMessage(mq, delivered.CorrelationId, conf.Broker.Exchange, conf.Broker.RoutingError, conf.Broker.Durable, body); e != nil {
+					log.Error("faild to publish message, reason: ", e)
+				}
```
Comment (pontus) [inline cmd/finalize/finalize.go:160]: Can't this be made a function in broker rather than having this much code for messages everywhere)?

Comment (jbygdell) [reply cmd/finalize/finalize.go:160]: I was thinking about it

**Review by pontus — APPROVED** [review]
I haven't actually verified but seems fine in principle, though I think it would very good to bring as much as possible into broker to lower code duplication._

**Review by blankdots — COMMENTED** [review]
small typo `errorus` should be `erroneous `

`// TODO fix resend mechanism` - an issue about this would track it better

**Review by blankdots — APPROVED** [review]
same comment as @pontus about code duplication

**Review by blankdots — APPROVED** [review]

**Pull request merged**